### PR TITLE
Auto map constructor getter setter

### PR DESCRIPTION
**Pull request by LambdAurora**
Attempts to auto map constructors parameters, getters and setters!

### CODE REVIEW

File: src/main/java/org/quiltmc/enigmaplugin/index/ConstructorParametersIndex.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+			// Search for every field assignation.
+			if (inst.getOpcode() == Opcodes.PUTFIELD) {
+				var fieldInst = (FieldInsnNode) inst;
+
+				if (!fieldInst.owner.equals(classNode.name)) continue; // The owner isn't this class.
+
+				var previousInst = fieldInst.getPrevious();
+
+				if (previousInst.getOpcode() >= Opcodes.ILOAD && previousInst.getOpcode() <= Opcodes.ALOAD) {
+					var loadInst = (VarInsnNode) previousInst;
```
Comment (ByMartrixx) [inline src/main/java/org/quiltmc/enigmaplugin/index/ConstructorParametersIndex.java:61]: What happens if the value is a local variable?

File: src/main/java/org/quiltmc/enigmaplugin/index/ConstructorParametersIndex.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+				if (instNode.owner.equals(classNode.name) && instNode.name.equals("<init>")) {
+					return true;
+				}
+			}
+		}
+
+		return false;
+	}
+
+	public record Entry(FieldEntry field, LocalVariableEntry constructorParam) {
```
Comment (ByMartrixx) [inline src/main/java/org/quiltmc/enigmaplugin/index/ConstructorParametersIndex.java:93]: Unused?

File: src/main/java/org/quiltmc/enigmaplugin/util/AsmUtil.java [hunk 3]
```diff
@@ -32,4 +38,81 @@ public static boolean maskMatch(int value, int... masks) {
...
+		if (Descriptors.getDescriptor(node).getArgumentDescs().size() != 1) return Optional.empty();
+		if (node.instructions.size() != 4) return Optional.empty();
+		if (node.instructions.get(0).getOpcode() != Opcodes.ALOAD) return Optional.empty();
+		if (node.instructions.get(3).getOpcode() != Opcodes.RETURN) return Optional.empty();
+		var putFieldNode = node.instructions.get(2);
+		if (putFieldNode.getOpcode() != Opcodes.PUTFIELD) return Optional.empty();
+
+		var fieldInsnNode = (FieldInsnNode) putFieldNode;
+
+		int expectedReturnOpcode = switch (fieldInsnNode.desc) {
```
Comment (ByMartrixx) [inline src/main/java/org/quiltmc/enigmaplugin/util/AsmUtil.java:102]: Just nitpicking - should be `expectedFieldOpcode`

**Review by ByMartrixx — APPROVED** [review]
I still have to test it in QM, but looks great so far!

File: src/main/java/org/quiltmc/enigmaplugin/index/ConstructorParametersIndex.java (hunk 1 above)
Comment (LambdAurora) [reply src/main/java/org/quiltmc/enigmaplugin/index/ConstructorParametersIndex.java:61]: Mmh, I'm not entirely sure.

**Pull request merged**